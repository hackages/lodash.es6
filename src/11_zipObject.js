/**
 * This method accepts two arrays, one of property identifiers and one of corresponding values
 */
export function zipObject(props, values) {}
