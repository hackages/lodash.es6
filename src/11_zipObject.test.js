import {zipObject} from './11_zipObject'

describe('zipObject', () => {
  it('should zip', () => {
    expect(zipObject(['a', 'b'], [1, 2])).toMatchSnapshot()
    expect(zipObject(['a'], [1, 2])).toMatchSnapshot()
    expect(zipObject(['a', 'b'], [1])).toMatchSnapshot()
    expect(zipObject(['hello'], [true])).toMatchSnapshot()
    expect(zipObject([32], [false])).toMatchSnapshot()
    expect(zipObject([true], ['or not'])).toMatchSnapshot()
  })
})
