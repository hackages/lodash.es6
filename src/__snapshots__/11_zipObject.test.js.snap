// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zipObject should zip 1`] = `
Object {
  "a": 1,
  "b": 2,
}
`;

exports[`zipObject should zip 2`] = `
Object {
  "a": 1,
}
`;

exports[`zipObject should zip 3`] = `
Object {
  "a": 1,
  "b": undefined,
}
`;

exports[`zipObject should zip 4`] = `
Object {
  "hello": true,
}
`;

exports[`zipObject should zip 5`] = `
Object {
  "32": false,
}
`;

exports[`zipObject should zip 6`] = `
Object {
  "true": "or not",
}
`;
